### PR TITLE
Add spatial filter to GeoJson layers

### PR DESCRIPTION
**Pull request by Clebal**
**MERGE THIS ONE BEFORE #246**

### CODE REVIEW

File: packages/react-api/src/hooks/dataFilterExtensionUtils.js [hunk 1]
```diff
@@ -55,34 +52,14 @@ function getUpdateTriggers(filtersWithoutTimeType, timeFilter) {
...
-    return result;
-  };
-}
-
 // The deck.gl DataFilterExtension accepts up to 4 values to filter.
 // We're going to use the first value for all filter except the time filter
 // that will be managed by the second value of the DataFilterExtension
 export function getDataFilterExtensionProps(filters = {}) {
-  const { filtersWithoutTimeType, timeFilter } = getFiltersByType(filters);
+  const [filtersWithoutTimeType, timeFilter] = getFiltersByType(filters);
```
Comment (padawannn) [inline packages/react-api/src/hooks/dataFilterExtensionUtils.js:59]: Remove getFilterValue break the builder compatibility and move this function to a hook makes it not possible to use it in builder

**Review by padawannn — CHANGES_REQUESTED** [review]

File: packages/react-api/src/hooks/useCartoLayerProps.js [hunk 2]
```diff
@@ -1,8 +1,9 @@
 import { MAP_TYPES, API_VERSIONS } from '@deck.gl/carto';
 import { useSelector } from 'react-redux';
-import useGeoJsonFeatures from './useGeoJsonFeatures';
-import useTilesetFeatures from './useTilesetFeatures';
-import { getDataFilterExtensionProps } from './dataFilterExtensionUtil';
+import useGeoJsonFeatures from './geojson/useGeoJsonFeatures';
+import useTilesetFeatures from './tileset/useTilesetFeatures';
+import { getDataFilterExtensionProps } from './dataFilterExtensionUtils';
```
Comment (padawannn) [inline packages/react-api/src/hooks/useCartoLayerProps.js:5]: Don't rename dataFilterExtensionUtil by dataFilterExtensionUtil`s`. It's break compatibility

**Review by padawannn — CHANGES_REQUESTED** [review]

File: packages/react-api/src/hooks/useCartoLayerProps.js [hunk 3]
```diff
@@ -28,25 +31,32 @@ export default function useCartoLayerProps({
 
   let props = {};
 
-  if (
+  const useMVT =
```
Comment (padawannn) [inline packages/react-api/src/hooks/useCartoLayerProps.js:34]: In c4r and cloud-native we try to don't use `MVT` for naming. Better useTileset

File: packages/react-core/src/filters/viewportFeaturesGeoJSON.d.ts [hunk 4]
```diff
@@ -1,8 +1,9 @@
-import { FeatureCollection } from 'geojson';
+import { FeatureCollection, Geometry } from 'geojson';
 import { Viewport, ViewportFeaturesResponse } from '../types';
 
 type ViewportFeaturesGeoJSONArgs = {
   geojson: FeatureCollection,
+  spatialFilterGeometry: Geometry,
```
Comment (padawannn) [inline packages/react-core/src/filters/viewportFeaturesGeoJSON.d.ts:6]: spatialFilterGeometry can be null, right? so it should be optional

**Review by padawannn — CHANGES_REQUESTED** [review]

File: packages/react-core/src/filters/viewportFeaturesGeoJSON.d.ts (hunk 4 above)
Comment (Clebal) [reply packages/react-core/src/filters/viewportFeaturesGeoJSON.d.ts]: Updated

File: packages/react-api/src/hooks/useCartoLayerProps.js (hunk 3 above)
Comment (Clebal) [reply packages/react-api/src/hooks/useCartoLayerProps.js]: Agree. Updated.

File: packages/react-api/src/hooks/useCartoLayerProps.js (hunk 2 above)
Comment (Clebal) [reply packages/react-api/src/hooks/useCartoLayerProps.js]: Updated.

File: packages/react-api/src/hooks/dataFilterExtensionUtils.js (hunk 1 above)
Comment (Clebal) [reply packages/react-api/src/hooks/dataFilterExtensionUtils.js]: Updated.

File: packages/react-api/__tests__/hooks/dataFilterExtensionUtils.test.js [hunk 5]
```diff
@@ -1,9 +1,9 @@
 import {
   getDataFilterExtensionProps,
   MAX_GPU_FILTERS
-} from '../../src/hooks/dataFilterExtensionUtil';
+} from '../../src/hooks/dataFilterExtensionUtils';
 
-describe('gpu-filter-utils', () => {
+describe('DataFilterExtensionUtils', () => {
```
[the remark this answers is not in the document]
Comment (Clebal) [reply packages/react-api/__tests__/hooks/dataFilterExtensionUtils.test.js]: Updated.

**Pull request closed without merging**